static/js/compose.js

```diff
@@ -642,14 +636,6 @@ exports.validation_error = function (error_type, stream_name) {
                 $("#stream_message_recipient_stream"),
             );
             return false;
-        case "not-subscribed": {
-            const sub = stream_data.get_sub(stream_name);
-            const new_row = render_compose_not_subscribed({
-                should_display_sub_button: sub.should_display_subscription_button,
-            });
-            compose_not_subscribed_error(new_row, $("#stream_message_recipient_stream"));
-            return false;
-        }
```

Comment:
Apart from removing this `case` block here, we also need to deal with places where `validation_error` is called with `error-type = 'not-subscribed' ` (like the `validate_stream_message` -> `validate_stream_message_address_info` -> `check_unsubscribed_stream_for_send` codepath)